test/checkoutput/README.md

```diff
@@ -0,0 +1,54 @@
+# checkoutput tool
+
+The checkoutput tool parses a text or markdown that has a command and the expected output sequence, runs the command and checks the output of the command against the expected output. 
+
+**Usage:**
+```
+<tool> -parsefile <path/to/file> -verbose
+
+-verbose is optional
+-parsefile should specify the path of the input markdown file.
+```
+
+The tool has 2 major functions
+
+1. Parsing and capturing the Run/Verify blocks
+2. Run and verify the output of the blocks captured in the previous step
+
+## Parsing
+The parsing is done using an Finite State Machine(FSM). The FSM has 2 states Start and Parsing. The initial state is Start. The parsing logic is summarized in the following table.
+
+| State | Input | Action | Output State
+|------ |-------|--------|-------------|
+| Start | Run Block | Record the command| Parsing|
+| Parsing | Run Block | Record the command| Parsing|
+| Parsing | Verify Block | Record the output | Start|
+
+An alternative is to use a markdown parser but that would help us only in the Action column and it would introduce external dependencies.
+
+### Example
+
+In the following example, the tool will parse the content of the "Run" block, runs the command. The output of the command is then checked against the text in "Expected output".
+
+Run:
+```console
+go run ./staging/src/k8s.io/component-base/logs/example/cmd/logger.go
```

Comment:
This works only because `logger.go` is self-contained. It does not work for binaries which are split into different files, because then the entire package has to be compiled and that doesn't work for `staging` (see comment above):
```
$ go run ./staging/src/k8s.io/component-base/logs/example/cmd
go run: no packages loaded from ./staging/src/k8s.io/component-base/logs/example/cmd
```

`go run .` works if we invoke commands inside the binary's directory.

Reply:
I've argued before that pre-built binaries should be used. I still see some value in that (for example, it ensures that binaries are tested exactly as they will get released), but perhaps "go run" is close enough. Let's keep it simple for now and use "go run".

Reply:
Do you have any examples of pre-built binaries that I can use in the tool ?

Reply:
Just run "make" and check under `_output`. But this isn't important right now. `go run .` will work fine for most cases, so let's start with that.